disnake/ext/commands/interaction_bot_base.py

```diff
@@ -237,11 +260,30 @@ def add_slash_command(self, slash_command: InvokableSlashCommand) -> None:
         if not isinstance(slash_command, InvokableSlashCommand):
             raise TypeError("The slash_command passed must be an instance of InvokableSlashCommand")
 
-        if slash_command.name in self.all_slash_commands:
-            raise CommandRegistrationError(slash_command.name)
-
         slash_command.body.localize(self.i18n)
-        self.all_slash_commands[slash_command.name] = slash_command
+        if slash_command.guild_ids:
+            for guild_id in slash_command.guild_ids:
+                if disnake.utils.get(
+                    self._all_app_commands,
+                    name=slash_command.name,
+                    type=slash_command.type,
+                    guild_id=guild_id,
+                ):
+                    raise CommandRegistrationError(slash_command.name, guild_id=guild_id)
+                self._all_app_commands[
+                    AppCommandMetadata(slash_command.name, guild_id, slash_command.type)
+                ] = slash_command
```

Comment:
What if we construct a named tuple here and use it as a key to check the existence? Like so:
```python
command_metadata = AppCommandMetadata(slash_command.name, guild_id, slash_command.type)
if command_metadata in self._all_app_commands:
    raise CommandRegistrationError(slash_command.name, guild_id=guild_id)
self._all_app_commands[command_metadata] = slash_command
```
Same for similar samples below